### PR TITLE
hw27

### PR DESCRIPTION
**Pull request by Troyan555**
complete

### CODE REVIEW

File: scr/DB/task2ALL.sql [hunk 1]
```diff
@@ -0,0 +1,57 @@
+use classicmodels;
+select e.firstName, e.lastName, o.city from employees e, offices o
```
Comment (SerhiiPasko) [inline scr/DB/task2ALL.sql:2]: There you have incorrect results. For every employee, you show two offices. Join should be written

File: scr/DB/task2ALL.sql [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+where o.city in ('Paris','Tokyo');
+
+select e.FirstName, e.LastName, count(c.contactFirstName) customers from employees e
+left join customers c on e.employeeNumber=c.salesRepEmployeeNumber
+group by e.FirstName, e.LastName
+order by e.FirstName asc;
+
+SELECT e2.firstName as Manager_FN, e2.lastName as Manager_LN, e1.firstName, e1.lastName FROM employees e1
+left join employees e2 on e2.employeeNumber = e1.reportsTo
+where e1.reportsTo = 1102;
```
Comment (SerhiiPasko) [inline scr/DB/task2ALL.sql:12]: What is `1102`? You have in task `Bondur Gerard`

File: scr/DB/task2ALL.sql [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+select e.FirstName, e.LastName, count(c.contactFirstName) customers from employees e
+left join customers c on e.employeeNumber=c.salesRepEmployeeNumber
+group by e.FirstName, e.LastName
+having customers <5
+order by customers desc;
+
+select e.employeeNumber, e.firstName, e.lastName, o.shippedDate from employees e
```
Comment (SerhiiPasko) [inline scr/DB/task2ALL.sql:20]: What about `DISTINCT`? We have duplicates now

File: scr/DB/task2ALL.sql [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+order by e.FirstName asc;
+
+select e.firstName, e.lastName, COUNT(distinct o.orderNumber) AS orderscount
+from employees e
+left join customers c on e.employeeNumber=c.salesRepEmployeeNumber
+left join orders o on c.customerNumber=o.customerNumber
+group by e.firstName,e.lastName
+order by orderscount desc;
+
+select e.firstName, e.lastName, COUNT(distinct o.orderNumber) AS orderscount
```
Comment (SerhiiPasko) [inline scr/DB/task2ALL.sql:33]: You may use something like:
```
WITH t (FirstName,LastName,orders) as
(SELECT e.firstName,e.lastName,count(o.orderNumber)
           from employees e 
           left join customers c on e.employeeNumber=c.salesRepEmployeeNumber
           left join orders o on c.customerNumber=o.customerNumber
           group by e.firstName,e.lastName)
Select * from t
where t.orders = (Select MIN(orders) from t)
or t.orders = (Select MAX(orders) from t);
```

**Review by SerhiiPasko — COMMENTED** [review]
Cool

File: scr/DB/task2ALL.sql (hunk 3 above)
Comment (Troyan555) [reply scr/DB/task2ALL.sql:20]: We have no duplicates when we run a query

**Pull request merged**